components/apimgt/org.wso2.carbon.apimgt.rest.api.service.catalog/src/main/java/org/wso2/carbon/apimgt/rest/api/service/catalog/OASYamlApi.java

```diff
@@ -0,0 +1,91 @@
+/*
+ *
+ *  Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package org.wso2.carbon.apimgt.rest.api.service.catalog;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.apimgt.api.APIManagementException;
+import org.wso2.carbon.apimgt.impl.RESTAPICacheConfiguration;
+import org.wso2.carbon.apimgt.impl.definitions.OAS3Parser;
+import org.wso2.carbon.apimgt.impl.utils.APIUtil;
+import org.wso2.carbon.apimgt.rest.api.util.utils.RestApiUtil;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.CacheControl;
+import javax.ws.rs.core.Response;
+import java.io.IOException;
+
+@Path("/oas.yaml")
+@Consumes({ "text/yaml" })
+@Produces({ "text/yaml" })
+@io.swagger.annotations.Api(value = "/oas.yaml", description = "the oas.yaml API")
+public class OASYamlApi {
+
+    private static final Log log = LogFactory.getLog(OASYamlApi.class);
+    private static final String LOCK_SERVICE_CATALOG_OPENAPI_DEF = "LOCK_SERVICE_CATALOG_OPENAPI_DEF";
+    private String openAPIDef = null;
+
+    /**
+     * Retrieves swagger definition of Service Catalog REST API and returns
+     * 
+     * @return swagger definition of Service Catalog REST API in yaml format
+     */
+    @GET
+    @Consumes({ "text/yaml" })
+    @Produces({ "text/yaml" })
+    @io.swagger.annotations.ApiOperation(value = "Get OpenAPI Definition", notes = "Get OpenAPI Definition of Service Catalog REST API.", response = Void.class)
+    @io.swagger.annotations.ApiResponses(value = {
+            @io.swagger.annotations.ApiResponse(code = 200, message = "OK.\nOpenAPI Definition is returned."),
+
+            @io.swagger.annotations.ApiResponse(code = 304, message = "Not Modified.\nEmpty body because the client has already the latest version of the requested resource."),
+
+            @io.swagger.annotations.ApiResponse(code = 406, message = "Not Acceptable.\nThe requested media type is not supported") })
+
+    public Response swaggerYamlGet() throws APIManagementException {
```

Comment:
Shall we change the method name also?

Reply:
fixed in c670fec543a
Thanks for the pointer